runtime/testenv/testenv.go

```diff
@@ -52,11 +53,23 @@ var (
 	env *envtest.Environment
 )
 
-func init() {
-	klog.InitFlags(nil)
+func initLogger() {
+	// NOTE: Set the klog flags on a separate FlagSet to avoid conflicting with
+	// any flags set by the application code. For example, an application
+	// importing glog would conflict with the flags set by klog, resulting in
+	// redeclaration of the same flags.
+	// This is based on klog's coexist with glog example, refer:
+	// https://github.com/kubernetes/klog/blob/f8e668dbaa5f6f0e6a5c24ffd7667263840d79ae/examples/coexist_glog/coexist_glog.go
+	// The klog flags set below are not available to the klogr logger below
```

Comment:
This, we have discovered, is not correct. `klogr` uses the public API of `klog`, which in turn makes use of the flagset to set itself up.

Reply:
Removed it with a statement about how to override this logger setup.